crates/symbolicator-service/src/services/sourcemap_lookup.rs

```diff
@@ -65,6 +83,14 @@ pub struct SourceMapLookup {
 
     artifacts: HashMap<String, CacheEntry<ByteView<'static>>>,
     sourcemaps: HashMap<String, CacheEntry<OwnedSourceMapCache>>,
+
+    /// Arbitrary files keyed by their [`FileKey`],
+    /// which is a combination of `abs_path` and `DebugId`.
+    files_by_key: HashMap<FileKey, CacheEntry<CachedFile>>,
+    /// SourceMaps by [`FileKey`].
+    sourcemaps_by_key: HashMap<FileKey, CacheEntry<OwnedSourceMapCache>>,
```

Comment:
Is it necessary to have both `sourcemaps_by_key` and `sourcemaps`? `sourcemaps_by_key` seems strictly more general.

Reply:
you are absolutely right! I rather want to experiment with these things in a clean-room so to say, without having to also change all the other code related to this.
I hope once this is done, `sourcemaps` and `artifacts` will just be removed completely.

Reply:
Cool, makes sense. So you would leave it as-is for this PR and simplify it in a follow-up?